### PR TITLE
[Merged by Bors] - perf(CharP/CharZero): scope simp theorems with weak keys

### PR DESCRIPTION
**Pull request by mattrobball**
These have keys like `HAdd.hAdd _ _ _ _ _ _` but are specific to characteristic two and zero.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mattrobball** [comment]
!bench

**Comment by MichaelStollBayreuth** [comment]
<b>General information:</b>
```text
lint:                                                  -89.941 * 10⁹ (-1.00%)
build:                                                 -119.89 * 10⁹ (-0.0963%)
```


<b>2 files got slower by at least 10⁹ instructions:</b>
```text
Mathlib.LinearAlgebra.TensorProduct.Graded.External:   +2.3693 * 10⁹ (+0.952%)
Mathlib.NumberTheory.RamificationInertia:              +2.2954 * 10⁹ (+0.757%)
```


<b>No file got slower by at least 10%.</b>

<details><summary><b>30 files got faster by at least 10⁹ instructions</b></summary>

```text
Mathlib.AlgebraicGeometry.EllipticCurve.Affine:        -6.4660 * 10⁹ (-3.46%)
Mathlib.Analysis.SpecialFunctions.Gaussian.FourierTransform:
                                                       -3.1198 * 10⁹ (-3.89%)
Mathlib.Geometry.Manifold.Algebra.LeftInvariantDerivation:
                                                       -2.7301 * 10⁹ (-1.49%)
Mathlib.Geometry.Manifold.Instances.Sphere:            -2.1862 * 10⁹ (-1.09%)
Mathlib.Analysis.InnerProductSpace.Projection:         -2.0243 * 10⁹ (-0.935%)
Mathlib.Algebra.Lie.Classical:                         -1.8201 * 10⁹ (-4.70%)
Mathlib.Algebra.Lie.Weights.Killing:                   -1.7934 * 10⁹ (-0.798%)
Mathlib.FieldTheory.Adjoin:                            -1.7735 * 10⁹ (-0.597%)
Mathlib.NumberTheory.Cyclotomic.Three:                 -1.6843 * 10⁹ (-3.48%)
Mathlib.AlgebraicGeometry.EllipticCurve.Jacobian:      -1.5992 * 10⁹ (-0.716%)
Mathlib.Algebra.Lie.Weights.Linear:                    -1.5968 * 10⁹ (-2.33%)
Mathlib.RingTheory.Kaehler.Basic:                      -1.5764 * 10⁹ (-0.419%)
Mathlib.LinearAlgebra.Orientation:                     -1.5491 * 10⁹ (-2.14%)
Mathlib.NumberTheory.Pell:                             -1.5461 * 10⁹ (-3.73%)
Mathlib.MeasureTheory.Integral.FundThmCalculus:        -1.4766 * 10⁹ (-1.71%)
Mathlib.Analysis.Calculus.UniformLimitsDeriv:          -1.3776 * 10⁹ (-2.80%)
Mathlib.LinearAlgebra.Matrix.Charpoly.Coeff:           -1.3320 * 10⁹ (-1.81%)
Mathlib.NumberTheory.Wilson:                           -1.3219 * 10⁹ (-19.9%)
Mathlib.LinearAlgebra.RootSystem.Basic:                -1.3137 * 10⁹ (-3.13%)
Mathlib.Analysis.InnerProductSpace.TwoDim:             -1.2953 * 10⁹ (-0.934%)
Mathlib.MeasureTheory.Function.LpSpace:                -1.2898 * 10⁹ (-0.528%)
Mathlib.Algebra.Lie.Abelian:                           -1.2686 * 10⁹ (-2.53%)
Mathlib.Geometry.Euclidean.Angle.Oriented.Basic:       -1.2672 * 10⁹ (-0.987%)
Mathlib.LinearAlgebra.CliffordAlgebra.Contraction:     -1.1982 * 10⁹ (-1.01%)
Mathlib.Analysis.InnerProductSpace.Orientation:        -1.1978 * 10⁹ (-2.25%)
Mathlib.Analysis.PSeries:                              -1.1518 * 10⁹ (-1.99%)
Mathlib.LinearAlgebra.Eigenspace.Basic:                -1.1407 * 10⁹ (-0.908%)
Mathlib.Geometry.Euclidean.Angle.Oriented.RightAngle:  -1.0833 * 10⁹ (-1.67%)
Mathlib.LinearAlgebra.CliffordAlgebra.Inversion:       -1.0501 * 10⁹ (-4.48%)
Mathlib.Analysis.SpecialFunctions.Integrals:           -1.0397 * 10⁹ (-1.28%)
```

</details>


<b>1 files got faster by at least 10%:</b>
```text
Mathlib.NumberTheory.Wilson:                                          -19.9%
```

**Review by MichaelStollBayreuth — APPROVED** [review]
LGTM

File: Mathlib/Algebra/CharZero/Lemmas.lean [hunk 1]
```diff
@@ -87,10 +87,10 @@ section
 
 variable {R : Type*} [NonAssocRing R] [NoZeroDivisors R] [CharZero R]
 
-@[simp] theorem neg_eq_self_iff {a : R} : -a = a ↔ a = 0 :=
+@[scoped simp] theorem CharZero.neg_eq_self_iff {a : R} : -a = a ↔ a = 0 :=
```
Comment (eric-wieser) [inline Mathlib/Algebra/CharZero/Lemmas.lean:90]: The old names here probably should have been deprecated.

Comment (mattrobball) [reply Mathlib/Algebra/CharZero/Lemmas.lean:90]: Yes. My mistake

**Pull request closed without merging**